examples/decomposition/apply_covid.py

```diff
@@ -0,0 +1,180 @@
+# Introduction
+# -----------------------
+# PARAFAC (CP) decomposition is extremely useful in dimensionality reduction, allowing us
+# to develop models that are both representative and compact while retaining crucial patterns
+# between subjects. Here, we provide an example of how it can be applied to biomedical research.
+#
+# Systems serology is a new technology that examines the antibodies from a patient's serum, aiming
+# to comprehensively profile the interactions between the antibodies and
+# [Fc receptors](https://en.wikipedia.org/wiki/Fc_receptor) alongside other types of immunological
+# and demographic data. Here, we will apply CP decomposition to a [COVID-19 system serology dataset]
+# (https://www.sciencedirect.com/science/article/pii/S0092867420314598).
+#
+# Details of this analysis as well as more in-depth biological implication can be found in
+# [this work](https://www.embopress.org/doi/full/10.15252/msb.202110243).
+#
+# We first import this dataset of a panel of COVID-19 patients:
+
+import numpy as np
+import pandas as pd
+from tensorly.datasets.imports import COVID19_data
+
+data = COVID19_data()
+
+##############################################################################
+# Apply CP decomposition to this dataset with Tensorly
+# -----------------------
+# Now we apply CP decomposition to this dataset.
+
+import tensorly as tl
+from tensorly.decomposition import parafac
+
+comps = np.arange(1, 7)
+CMTFfacs = [parafac(data.tensor, cc, tol=1e-10, n_iter_max=1000,
+                    linesearch=True, orthogonalise=2) for cc in comps]
+
+##############################################################################
+# To evaluate how well CP decomposition explains the variance in the dataset, we plot the percent
+# variance reconstructed (R2X) for a range of ranks.
+
+def calcR2X(tFac, tIn=None):
+    tMask = np.isfinite(tIn)
+    vTop = np.sum(np.square(tl.cp_to_tensor(tFac) * tMask - np.nan_to_num(tIn)))
+    vBottom = np.sum(np.square(np.nan_to_num(tIn)))
+    return 1.0 - vTop / vBottom
+
+import matplotlib
+from matplotlib import pyplot as plt
+
+fig1 = plt.figure()
+ax1 = fig1.add_axes([0,0,1,1])
+CMTFR2X = np.array([calcR2X(f, data.tensor) for f in CMTFfacs])
+print(CMTFR2X)
+ax1.plot(comps, CMTFR2X, "bo")
+ax1.set_ylabel("R2X")
+ax1.set_xlabel("Number of Components")
+ax1.set_xticks([x for x in comps])
+ax1.set_xticklabels([x for x in comps])
+ax1.set_ylim(0, 1)
+ax1.set_xlim(0.0, np.amax(comps) + 0.5)
+
+##############################################################################
+# Build a Logistic Regression classifier to predict severe versus deceased COVID-19 patients
+# -----------------------
+# An important function of CP decomposition here is to reduce data while keeping significant
+# information.
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.metrics import roc_curve, roc_auc_score
+from sklearn.model_selection import KFold
```

Comment:
That's going to crash the current doc as we don't install scikit-learn. We can add it if this is crucial to have.

Reply:
This is mostly for making predictions and plotting the ROC curve. It would be an interesting plot to show, especially for how tensor method can be used in ML context, but this is not essential to the function.

Reply:
Also, I don't know if seaborn is a dependency, but we use them to plot too so hopefully it's not a problem.

Reply:
@cyrillustan I think we can just plot the factors and then point out there is a difference between severe / deceased.

Regarding seaborn, if it's not too difficult to just use matplotlib, that would limit the number of necessary packages.

Reply:
Okay, removed the ROC curve and changed all plots in matplotlib. Just a reminder that even matplotlib is not a dependency currently.

Reply:
Thanks! Matplotlib is a dependency for the doc.